### ISSUE TITLE
Inline python re modifiers not working

### ISSUE THREAD

**Issue by SteveBarnes-BH** [firasdib/Regex101#1698]
## Bug Description
The regex 
`#define\s+(?i:CONFIGXML_HEADER)`
reports:
```
(? Incomplete group structure
) Incomplete group structure
```
However, it is a valid regular expression in python 3.9 and possibly others meaning that I need `#define ` as case sensitive but `CONFIGXML_HEADER` case insensitive.

![image](https://user-images.githubusercontent.com/88222167/148973471-4ee0d4fc-2765-42b0-8bb7-cf555ceb1d4b.png)
### Reproduction steps
Paste the above regex into the regex field on the site.

### Expected Outcome
Partially case sensitive regex.

### Browser
Chrome

### OS
Windows 10

**Comment by firasdib** [comment]
This is a longer standing issue, the website caters for Python 2.7, which is very outdated at this point. I will have to rework it completely to support Python 3+ ASAP.

**Comment by SteveBarnes-BH** [comment]
In this context it is probably worth mentioning that all _official_ support for Python 2.x ended 01/01/2020.

**Comment by firasdib** [comment]
@SteveBarnes-BH Is there a writeup somewhere outlining the regex differences between 2.7 and 3.x?

**Comment by SteveBarnes-BH** [comment]
https://docs.python.org/3/library/re.html has:

> (?aiLmsux-imsx:...)
> (Zero or more letters from the set 'a', 'i', 'L', 'm', 's', 'u', 'x', optionally followed by '-' followed by one or more letters from the 'i', 'm', 's', 'x'.) The letters set or remove the corresponding flags: re.A (ASCII-only matching), re.I (ignore case), re.L (locale dependent), re.M (multi-line), re.S (dot matches all), re.U (Unicode matching), and re.X (verbose), for the part of the expression. (The flags are described in Module Contents.)
> 
> The letters 'a', 'L' and 'u' are mutually exclusive when used as inline flags, so they can’t be combined or follow '-'. Instead, when one of them appears in an inline group, it overrides the matching mode in the enclosing group. In Unicode patterns (?a:...) switches to ASCII-only matching, and (?u:...) switches to Unicode matching (default). In byte pattern (?L:...) switches to locale depending matching, and (?a:...) switches to ASCII-only matching (default). This override is only in effect for the narrow inline group, and the original matching mode is restored outside of the group.
> 
> New in version 3.6.
> 
> Changed in version 3.7: The letters 'a', 'L' and 'u' also can be used in a group.

The [How To](https://docs.python.org/3/howto/regex.html#regex-howto) is a useful resource as well.

There is also the significant difference that you can have string, (i.e. Unicode), or byte regular expressions and also targets and that the 2 don't mix, i.e. re.findall("Fred", b"Fred") will cause a error, (`TypeError: cannot use a string pattern on a bytes-like object`), but I would suggest this is probably best just being a comment on your site rather than trying to deal with it.

**Comment by thesuperzapper** [comment]
@firasdib have you managed to make any progress on this issue?

Either way, the name of the "Python" flavor should probably be "Python 2.7", to make sure users understand that Python 3 syntax is not supported.

**Comment by weallcock** [comment]
I would agree with the statement that you should make it very clear that this is python 2.7.  I will also just drop another 2.7 vs 3.9 difference in here for a test whenever you make the switch.  I entered this regex on your site: `\w*apple\b?` and it threw an error on the ? saying `The preceding token is not quantifiable`.  However, that regex is valid in 3.9.  Here is a tiny example:
```
# Find the words that end in apple
string = r'apple orange pear snapple pineapple durian'
# 0 or more characters in the word character class before apple with a word boundary after the e, non-greedy.
# 0 takes care of apple, more takes care of snapple and pineapple.
regex = '\w*apple\b?'
print(re.findall(regex,string))

['apple', 'snapple', 'pineapple']
```
I will also note that your site in python2.7 matched without the ?.  Without the ? you get not match, though I don't understand why.  That is actually why I came to your site.  I expected to get `'apple orange pear snapple pineapple'`.  I am using Python 3.9.13 from anaconda.  Thanks for the web site.  I think it is great.

**Comment by firasdib** [comment]
Will be fixed in the new version
